cruise-control/src/yaml/endpoints/removeDisks.yaml

```diff
@@ -0,0 +1,69 @@
+RemoveDisksEndpoint:
+  post:
+    operationId: removeDisks
+    summary: Move all replicas from the specified disk to other disks of the same broker.
+    parameters:
+      - name: dryrun
+        in: query
+        description: Whether to dry-run the request or not.
+        schema:
+          type: boolean
+          default: true
+      - name: stop_ongoing_execution
+        in: query
+        description: Whether to stop the ongoing execution (if any) and start executing the given request.
+        schema:
+          type: boolean
+          default: false
+      - name: reason
+        in: query
+        description: Reason for request.
+        schema:
+          type: string
+        example: "Balance disk utilization across all brokers in the cluster."
+      - name: brokerid_and_logdirs
+        in: query
+        description: List of broker id and logdir pair to be demoted in the cluster.
+        schema:
+          type: object
+          additionalProperties:
+            type: array
+            items:
+              type: string
+        required: true
+        example: 101-/tmp/kafka-logs-1,101-/tmp/kafka-logs-2
```

Comment:
How did you handle if the broker/logDir doesn't exist?

Reply:
Hi @CCisGG!
1. The remove broker endpoint marks the broker as dead, however we do not need to mark the whole broker as dead, only some of its disks.<br/>
I saw that there is an enum for disk states which contains the DEAD value, however when generating the cluster model at the start of the runnable (`clusterModel = _kafkaCruiseControl.clusterModel(...)`) the broker disk configuration is always read from the capacity configuration file. So even if the disk is marked as dead during the remove disks call, at a following rebalance call the cluster model will be read from the capacity file and the disk will have a healthy state. Please correct me if I am wrong.<br/>
I looked into this when trying to find a way to stop further rebalances to use the removed disks (until they are specifically removed from the Kafka broker configuration), but changing how the cluster model is read or how the disks are marked as dead seemed a bit too complex and could potentially affect other functionalities.<br/><br/>
2. Indeed the broker/logDir combination is validated in `RemoveDisksRunnable.checkCanRemoveDisks`. This goal should not be executed outside of a remove disks call. To prevent being called from the rebalance endpoint at the start of the`DiskRemovalGoal.optimize` method a call to `DiskRemovalGoal.sanityCheckOptimizationOptions` is made and throws an exception if the optimization was triggered by a goal violation. If this is not the way to prevent the call from the rebalance endpoint please tell.